### ISSUE TITLE
Error providing permissions with kotlin 1.7.20

### ISSUE THREAD

**Issue by pauminku** [icerockdev/moko-permissions#66]
I managed to configure moko-permissions library in my project and is working fine in iOS and Android for Gallery Permission, but when I try to do the same with Remote notifications permissions I'm receiving the error below in iOS (in Android is working fine):
The same error is throw either in `getPermissionState` and `providePermission()`

```
Uncaught Kotlin exception: kotlin.IllegalArgumentException: Failed requirement.
    at 0   myApp                          0x1033257e3        kfun:kotlin.Throwable#<init>(kotlin.String?){} + 123 
    at 1   myApp                          0x10331da4b        kfun:kotlin.Exception#<init>(kotlin.String?){} + 119 
    at 2   myApp                          0x10331dd17        kfun:kotlin.RuntimeException#<init>(kotlin.String?){} + 119 
    at 3   myApp                          0x10331e237        kfun:kotlin.IllegalArgumentException#<init>(kotlin.String?){} + 119 
    at 4   myApp                          0x1033351b3        kfun:kotlin.native.concurrent.Continuation1#invoke(1:0){} + 903 
    at 5   myApp                          0x1033352c3        kfun:kotlin.native.concurrent.Continuation1#$<bridge-UNNN>invoke(1:0){}(1:0){}kotlin.Any + 123 
    at 6   myApp                          0x103608183        _6465762e696365726f636b2e6d6f6b6f3a7065726d697373696f6e73_knbridge97 + 251 
    at 7   libdispatch.dylib                   0x1059946d3        _dispatch_call_block_and_release + 31 
    at 8   libdispatch.dylib                   0x1059963b3        _dispatch_client_callout + 19 
    at 9   libdispatch.dylib                   0x10599e53f        _dispatch_lane_serial_drain + 987 
    at 10  libdispatch.dylib                   0x10599f2c3        _dispatch_lane_invoke + 479 
    at 11  libdispatch.dylib                   0x1059abe1f        _dispatch_workloop_worker_thread + 915 
    at 12  libsystem_pthread.dylib             0x1f37230f3        _pthread_wqthread + 287 
    at 13  libsystem_pthread.dylib             0x1f3722e93        start_wqthread + 7 
dyld4 config: DYLD_LIBRARY_PATH=/usr/lib/system/introspection DYLD_INSERT_LIBRARIES=/Developer/usr/lib/libBacktraceRecording.dylib:/Developer/usr/lib/libMainThreadChecker.dylib:/Developer/Library/PrivateFrameworks/DTDDISupport.framework/libViewDebuggerSupport.dylib
```

**Comment by pauminku** [comment]
Correction: it's happening in other Permissions, not just Remote Notifications permission.

**Comment by Alex009** [comment]
can you reproduce this error in sample app from this repository?

**Comment by pauminku** [comment]
Sorry for the late response. I was not able to reproduce the error in the sample. I'm using my own ViewModel implementation (not MOKO viewmodel) could this be the problem?

**Comment by pauminku** [comment]
After some more testes, I've managed to reproduce the problem in the sample app. 
You just need to change the kotlin version to `kotlinVersion = "1.7.20"` and it will happen exactly the same error.
So I guess it's related to the new memory management: https://kotlinlang.org/docs/native-memory-manager.html
Is there any workaround?

**Comment by codlab** [comment]
I'm currently facing the same issue and also have 1.7.20+ set in my project. Is there a workaround for this ?

**Comment by Alex009** [comment]
will be released in 0.15.0
